cpp/benchmarks/pairwise_linestring_distance.cu

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+#include <nvbench/nvbench.cuh>
+
+#include <cuspatial/detail/iterator.hpp>
+#include <cuspatial/experimental/linestring_distance.cuh>
+#include <cuspatial/experimental/type_utils.hpp>
+
+#include <rmm/device_vector.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <memory>
+
+namespace cuspatial {
```

Comment:
Not sure why these functions are being added to the cuSpatial namespace?

Reply:
Good catch. The benchmarks shouldn't be counted in the library namespace because they are considered external code that treats the target API as a black box. I think it would be helpful to mention this in https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/docs/BENCHMARKING.md

Reply:
Updated.

Reply:
Did you open an issue for adding a namespace mention to BENCHMARKING.md?